pkg/ssh/ssh.go

```diff
@@ -18,9 +20,56 @@ import (
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 )
 
+// LoadKey loads the SSH public key specified in NodeGroupSSH and returns it. The key should be specified
+// in only one way: by name (for a key existing in EC2), by path (for a key in a local file)
+// or by its contents (in the config-file). It also assumes that if ssh is enabled (SSH.Allow
+// == true) then one key was specified
+func LoadKey(sshConfig *api.NodeGroupSSH, clusterName, nodeGroupName string, ec2API ec2iface.EC2API) (string, error) {
+	if sshConfig.Allow == nil || *sshConfig.Allow == false {
+		return "", nil
+	}
+
+	switch {
+
+	// Load Key by content
+	case sshConfig.PublicKey != nil:
+		keyName, err := LoadKeyByContent(sshConfig.PublicKey, clusterName, nodeGroupName, ec2API)
+		if err != nil {
+			return "", err
+		}
+		return keyName, nil
+
+		// Use key by name in EC2
+	case sshConfig.PublicKeyName != nil && *sshConfig.PublicKeyName != "":
+		if err := CheckKeyExistsInEC2(*sshConfig.PublicKeyName, ec2API); err != nil {
+			return "", err
+		}
+		logger.Info("using EC2 key pair %q", *sshConfig.PublicKeyName)
+		return *sshConfig.PublicKeyName, nil
+
+		// Local ssh key file
+	case file.Exists(*sshConfig.PublicKeyPath):
+		keyName, err := LoadKeyFromFile(*sshConfig.PublicKeyPath, clusterName, nodeGroupName, ec2API)
+		if err != nil {
+			return "", err
+		}
+		return keyName, nil
+
+		// A keyPath, when specified as a flag, can mean a local key (checked above) or a key name in EC2
+	default:
+		err := CheckKeyExistsInEC2(*sshConfig.PublicKeyPath, ec2API)
+		if err != nil {
+			return "", err
+		}
+		logger.Info("using EC2 key pair %q", sshConfig.PublicKeyName)
+		return *sshConfig.PublicKeyPath, nil
+	}
+
+}
+
 // LoadKeyFromFile loads and imports a public SSH key from a file provided a path to that file.
 // returns the name of the key
-func LoadKeyFromFile(filePath, clusterName, ngName string, provider api.ClusterProvider) (string, error) {
+func LoadKeyFromFile(filePath, clusterName, ngName string, ec2API ec2iface.EC2API) (string, error) {
```

Comment:
Will we be able to inject a mock here if we use `ec2iface.EC2API`? I would think so but can you please double check?

Reply:
Yes, we do generate mocks for the EC2API interface (`*mocks.EC2API`). I have also used that mock in this PR